src/lib/import_export/binary/binary_writer.hpp

```diff
@@ -223,6 +224,38 @@ class BinaryWriter {
   template <typename T>
   static void _write_segment(const LZ4Segment<T>& lz4_segment, bool column_is_nullable, std::ofstream& ofstream);
 
+  /**
+   * FSSTSegments are dumped with the following layout:
+   *
+   * Description                 | Type                                | Size in bytes
+   * --------------------------------------------------------------------------------------------------------
+   * Encoding Type               | EncodingType                        | 1
+   * Attribute vector compr. ID. | CompressedVectorTypeID              | 1
+   * Compressed values size      | uint32_t                            | 4
+   * Compressed values           | vector<uint32_t>                    | Compressed values size * 1
+   * Vector compress. bit width  | uint8_t                             | 1
+   * Offset values               | uint8_t                             | (Rows + 1) * (vector compr. bit width) / 8
+   *                                                                     rounded up to next multiple of word (8 byte)
+   * Offset values               | uint(8|16|32)_t                     | (Rows + 1) * width of offset vector
+   * Reference offsets size      | uint32_t                            | 4
+   * Reference offsets           | vector<uint64_t>                    | 8 * 8
```

Comment:
Is that really a uint64_t? Isn't the used compressed bitpacking vector limited to uint32_t?

Reply:
The reference offsets are the 8 intermediate values for the offsets, which are saved in order to minimize the maximum element in the offsets vector, which is then being compressed with bit packing. Therefore, we use `uint64_t` for reference offsets, since those values are aggregated across all string lengths and thus can be potentially quite large.